mhkit/wave/resource.py

```diff
@@ -684,3 +686,259 @@ def func(kk):
     k = pd.DataFrame(k0, index=f, columns=['k'])
     
     return k
+
+def environmental_contour(x1, x2, dt, period, PCA=None, size_bin=250, 
+                          nb_steps=1000, return_PCA=False):
+    '''    
+    Calculates environmental contours of extreme sea 
+    states using the improved joint probability distributions 
+	with the inverse first-order reliability method (IFORM) 
+    probability for the desired return period (period). Given the 
+    period of interest a circle of iso-probability is created in the
+    in the PCA joint probability (x1, x2) reference frame.  
+    Using the joint probability value the CDF of the marginal 
+    distribution is used to find the quantile of each component. 
+    Finally, using the improved PCA methodology
+    the component 2 contour lines are calculated from component 1 using 
+    the relationships defined in _principal_component_analysis.	
+
+    Eckert-Gallup, A. C., Sallaberry, C. J., Dallman, A. R., & 
+    Neary, V. S. (2016). Application of principal component 
+    analysis (PCA) and improved joint probability distributions to 
+    the inverse first-order reliability method (I-FORM) for predicting 
+    extreme sea states. Ocean Engineering, 112, 307-319.
+
+    Parameters
+    ----------
+    x1: array like
+        Component 1 data
+    x2: array like
+        Component 2 data        	
+    dt : float
+        x1 and x2 sample rate (seconds) 
+    period : np.array
+        Desired return period (years) for calculation of environmental
+        contour, can be a scalar or a vector. If vector, must be an n by 1 array. 
+    PCA: dict (optional)
+	    properties returned from _principal_component_analysis
+    size_bin : float (optional)
+        Data points in each bin 		
+    nb_steps : int (optional)
+        Discretization of the circle in the normal space used for
+        IFORM calculation.
+	return_PCA: boolean (optional)
```

Comment:
Similar comment here, we should not reference a hidden function.

Reply:
Agreed I will remove.